pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -105,12 +112,17 @@ func (a *index) constructService(svc *v1.Service) *workloadapi.Service {
 		})
 	}
 
+	addresses, err := slices.MapErr(getVIPs(svc), a.toNetworkAddress)
```

Comment:
how could this happen

Reply:
tbh I'm not 100% sure but if it fails I would much rather reject the service than to panic. It's unacceptable to panic on invalid inputs coming from external sources